File: HW6-1_multiple-option/109590020_HW6-1/app/src/main/java/com/example/hw6_1/MainActivity.java
package com.example.hw6_1;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.CheckBox;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    private String mOrderMessage;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);


    }

    public void displayToast(String message) {
        Toast.makeText(getApplicationContext(), "Toppings: "+message,
                Toast.LENGTH_SHORT).show();
    }

    public void showtext(View view) {
        mOrderMessage="";
        CheckBox c1= findViewById(R.id.o1);
        CheckBox c2= findViewById(R.id.o2);
        CheckBox c3= findViewById(R.id.o3);
        CheckBox c4= findViewById(R.id.o4);
        CheckBox c5= findViewById(R.id.o5);
        if(c1.isChecked())
            mOrderMessage+="*"+ c1.getText().toString();
        if(c2.isChecked())
            mOrderMessage+="*"+ c2.getText().toString();
        if(c3.isChecked())
            mOrderMessage+="*"+ c3.getText().toString();
        if(c4.isChecked())
            mOrderMessage+="*"+ c4.getText().toString();
        if(c5.isChecked())
            mOrderMessage+="*"+ c5.getText().toString();
        displayToast(mOrderMessage);
    }

}

File: HW10-2_websourcecatcher/109590020_HW10-2/app/src/main/java/com/example/websourcefetcher/MainActivity.kt
package com.example.websourcefetcher

import android.os.Bundle
import android.view.MenuInflater
import android.widget.Button
import android.widget.EditText
import android.widget.PopupMenu
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import androidx.loader.app.LoaderManager
import androidx.loader.content.Loader

class MainActivity : AppCompatActivity(), LoaderManager.LoaderCallbacks<String> {


    private lateinit var httpTextView: TextView
    private lateinit var urlText: EditText
    private lateinit var pageSource: TextView
    private lateinit var Button: Button

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        httpTextView = findViewById(R.id.http_textview)
        httpTextView.setOnClickListener { view ->
            val popup = PopupMenu(this, view)
            val inflater: MenuInflater = popup.menuInflater
            inflater.inflate(R.menu.http_menu, popup.menu)

            popup.setOnMenuItemClickListener { item ->
                when (item.itemId) {
                    R.id.http -> httpTextView.text = getString(R.string.http_text)
                    R.id.https -> httpTextView.text = getString(R.string.https_text)
                }
                true
            }
            popup.show()
        }
        urlText = findViewById(R.id.url)
        Button = findViewById(R.id.button)
        Button.setOnClickListener {
            LoaderManager.getInstance(this).restartLoader(0, null, this)
            pageSource.text = "Loading ..."
        }
        pageSource = findViewById(R.id.page_source)
    }
    override fun onCreateLoader(id: Int, args: Bundle?): Loader<String> {
        val url = httpTextView.text.toString() + urlText.text.toString()
        return HtmlLoader(this, url)
    }

    override fun onLoadFinished(loader: Loader<String>, data: String?) {
        pageSource.text = data
    }
    override fun onLoaderReset(loader: Loader<String>) {}
}